game-core/src/test/resources/required-op-codes.txt

```diff
@@ -0,0 +1,134 @@
+3,games.strategy.engine.chat.IChatChannel,speakerAdded,org.triplea.domain.data.ChatParticipant
```

Comment:
Is there any way to document the structure and contents of this file and how it was generated? If we somehow added a new remote method, will it be easy for someone to update this (and to know to update it?)

Reply:
If you're searching for an inline way to document this file there probably isn't a good one.
CSV is a format that doesn't support comments unfortunately.

But if we documented this file in the test case via a javadoc that would work.
Basically it's just a list of method descriptors and their appropriate methods.
Updating this list ist simple too, just add another line below that matches the scheme:
```
op-code:full-interface-name:method-name:method-param-type-0:...;method-param-type-n
```

Reply:
I like the suggestion. I think we probably could use a `docs/` file somewhere to describe the remote method interface mechanism and perhaps give a check-list of things you need to update when adding a remote method. This way we preserve context of the moving pieces, some code examples even how the remote method invocation works and the differences between hub invoke, broadcast and point to point would all be great as well (perhaps those items can come later as we iterate on the documentation, for now probably most important to have a checklist of what you would need to touch if you added a remote method invocation so we can be a bit more confident that this test will be updated going forward)